common/lib/xmodule/xmodule/modulestore/tests/test_semantics.py

```diff
@@ -259,6 +260,24 @@ def test_delete(self, block_type):
         self.assertCourseDoesntPointToBlock(block_usage_key)
         self.assertBlockDoesntExist(block_usage_key)
 
+    def test_course_summaries(self):
+        def verify_course_summery_fields(course_summary):
+            """ Verify that every `course_summary` object has all the required fields """
+            expected_fields = CourseSummary.course_info_fields + ['id', 'location']
+            return all([hasattr(course_summary, field) for field in expected_fields])
+
+        with self.store.branch_setting(ModuleStoreEnum.Branch.published_only):
+            course_summaries = self.store.get_course_summaries()
+
+            # Verify course summaries
+            self.assertEqual(len(course_summaries), 1)
+
+            # Verify that `course_summaries` have all the required attributes.
+            self.assertTrue(all(verify_course_summery_fields(course_summary) for course_summary in course_summaries))
+
+            # Verify fetched accessible courses list is a list of CourseSummery instances
+            self.assertTrue(all(isinstance(course, CourseSummary) for course in course_summaries))
+
```

Comment:
I think it might make sense to test the behavior of `get_course_summaries` if you use the `draft_preferred` `branch_setting` as well.